sympy/combinatorics/fp_groups.py

```diff
@@ -135,15 +135,39 @@ def order(self, strategy="relator_based"):
         2
 
         """
+        from sympy.core import S
         if self._order != None:
             return self._order
         if self._coset_table != None:
             self._order = len(self._coset_table.table)
         else:
-            self._coset_table = self.coset_enumeration([], strategy)
-            self._order = len(self._coset_table.table)
+            if self._is_infinite():
+                self._order = S.Infinity
+            else:
+                self._coset_table = self.coset_enumeration([], strategy)
+                self._order = len(self._coset_table.table)
         return self._order
 
+    def _is_infinite(self):
+        '''
+        Test if the group is infinite. Return `True` if the test succeeds
+        and `None` otherwise
+
+        '''
+        # Abelianisation test: check is the abelianisation is infinite
+        abelian_rels = []
+        from sympy.matrices import Matrix
```

Comment:
This could probably be `from sympy.polys.solvers import RawMatrix as Matrix`. Then elements of `ZZ` would not be converted to Integers.